libcsm/tests/hsm/test_xnames.py

```diff
@@ -0,0 +1,88 @@
+#
+#  MIT License
+#
+#  (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a
+#  copy of this software and associated documentation files (the "Software"),
+#  to deal in the Software without restriction, including without limitation
+#  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#  and/or sell copies of the Software, and to permit persons to whom the
+#  Software is furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included
+#  in all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+#  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+#  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+#  OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Tests for the hsm get_xnames_by_subrole submodule.
+"""
+
+import pytest
+import mock
+
+from libcsm import api
+from libcsm.hsm import api as hsmApi
+from  libcsm.hsm import xnames
+
+class MockHTTPResponse:
+    def __init__(self, data, status_code):
+        self.json_data = data
+        self.status_code = status_code
+
+    def json(self):
+        return self.json_data
+
+class TestXnames:
+
+    @mock.patch('libcsm.api.Auth', spec=True)
```

Comment:
Since this is patched for every function, this decorator could move to the class itself.

```python
@mock.patch('libcsm.api.Auth', spec=True)
class TestXnames:
```